include/openssl/x509.h

```diff
@@ -466,12 +466,13 @@ int i2d_PUBKEY_bio(BIO *bp, const EVP_PKEY *pkey);
 EVP_PKEY *d2i_PUBKEY_bio(BIO *bp, EVP_PKEY **a);
 
 DECLARE_ASN1_DUP_FUNCTION(X509)
+DECLARE_ASN1_DUP_FUNCTION(X509_ALGOR)
 DECLARE_ASN1_DUP_FUNCTION(X509_ATTRIBUTE)
-DECLARE_ASN1_DUP_FUNCTION(X509_EXTENSION)
 DECLARE_ASN1_DUP_FUNCTION(X509_CRL)
-DECLARE_ASN1_DUP_FUNCTION(X509_REVOKED)
+DECLARE_ASN1_DUP_FUNCTION(X509_EXTENSION)
+DECLARE_ASN1_DUP_FUNCTION(X509_PUBKEY)
 DECLARE_ASN1_DUP_FUNCTION(X509_REQ)
-DECLARE_ASN1_DUP_FUNCTION(X509_ALGOR)
+DECLARE_ASN1_DUP_FUNCTION(X509_REVOKED)
```

Comment:
Maybe some unit tests would be good - coverage is already bad enough :)

Reply:
Are you serious? Or is this just your tit-for-tat response to my last overly pedantic comments? ;-)

Reply:
up to you - at some point coverage will be looked at.. maybe..

Reply:
I agree with you that the ASN1 parser could need some more covering by unit tests.  But I am not the person who would be able to implement them.  Also, I am already punished enough with all those missing accessors and const incorrectnesses encountered while porting code from 1.0.2 to 1.1.1.